code/datums/components/edible.dm

```diff
@@ -0,0 +1,238 @@
+		else if(fullness > (600 * (1 + eater.overeatduration / 2000)))	// The more you eat - the more you can eat
+			eater.visible_message("<span class='warning'>[eater] cannot force any more of \the [parent] to go down [eater.p_their()] throat!</span>", "<span class='warning'>You cannot force any more of \the [parent] to go down your throat!</span>")
+			return FALSE
+	else //If you're feeding it to someone else.
+		if(isbrain(eater))
+			to_chat(feeder, "<span class='warning'>[eater] doesn't seem to have a mouth!</span>")
+			return
+		if(fullness <= (600 * (1 + eater.overeatduration / 1000)))
+			eater.visible_message("<span class='danger'>[feeder] attempts to feed [eater] [parent].</span>", \
+									"<span class='userdanger'>[feeder] attempts to feed you [parent].</span>")
+		else
+			eater.visible_message("<span class='warning'>[feeder] cannot force any more of [parent] down [eater]'s throat!</span>", \
+									"<span class='warning'>[feeder] cannot force any more of [parent] down your throat!</span>")
+			return FALSE
+		if(!do_mob(feeder, eater)) //Wait 3 seconds before you can feed
+			return
+		log_combat(feeder, eater, "fed", owner.reagents.log_list())
+		eater.visible_message("<span class='danger'>[feeder] forces [eater] to eat [parent]!</span>", \
+									"<span class='userdanger'>[feeder] forces you to eat [parent]!</span>")
+
+	TakeBite(eater, feeder)
+	return COMPONENT_ITEM_NO_ATTACK
+
+///This function lets the eater take a bite and transfers the reagents to the eater.
+/datum/component/edible/proc/TakeBite(mob/living/eater, mob/living/feeder)
+	var/atom/owner = parent
+
+	if(!owner.reagents)
+		return FALSE
+	if(eater.satiety > -200)
+		eater.satiety -= junkiness
+	playsound(eater.loc,'sound/items/eatfood.ogg', rand(10,50), TRUE)
+	if(owner.reagents.total_volume)
+		SEND_SIGNAL(parent, COMSIG_FOOD_EATEN, eater, feeder)
+		var/fraction = min(bite_consumption / owner.reagents.total_volume, 1)
+		owner.reagents.trans_to(eater, bite_consumption, transfered_by = feeder, method = INGEST)
+		bitecount++
+		On_Consume(eater)
+		checkLiked(fraction, eater)
+
+		//Invoke our after eat callback if it is valid
+		if(after_eat)
+			after_eat.Invoke(eater, feeder)
+
+		return TRUE
+
+///Checks whether or not the eater can actually consume the food
+/datum/component/edible/proc/CanConsume(mob/living/eater, mob/living/feeder)
+	if(!iscarbon(eater))
+		return FALSE
+	var/mob/living/carbon/C = eater
+	var/covered = ""
+	if(C.is_mouth_covered(head_only = 1))
+		covered = "headgear"
+	else if(C.is_mouth_covered(mask_only = 1))
+		covered = "mask"
+	if(covered)
+		var/who = (isnull(feeder) || eater == feeder) ? "your" : "[eater.p_their()]"
+		to_chat(feeder, "<span class='warning'>You have to remove [who] [covered] first!</span>")
+		return FALSE
+	return TRUE
+
+///Check foodtypes to see if we should send a moodlet
+/datum/component/edible/proc/checkLiked(var/fraction, mob/M)

```

Comment:
Make a signal for something being eaten and put this code on the mood component

Reply:
I discussed this a bit with you on discord and I'm still not sure how to make sure this can be correcly overriden on special cases such as noone liking specific food or specific food being liked more by jobs.

Reply:
needs a response ninja

Reply:
eh this is fine I guess